src/wallet/init.cpp

```diff
@@ -185,51 +185,23 @@ bool VerifyWallets()
 
     uiInterface.InitMessage(_("Verifying wallet(s)..."));
 
-    // Keep track of each wallet absolute path to detect duplicates.
-    std::set<fs::path> wallet_paths;
+    std::vector<std::string> wallet_files = gArgs.GetArgs("-wallet");
 
-    for (const std::string& walletFile : gArgs.GetArgs("-wallet")) {
-        if (boost::filesystem::path(walletFile).filename() != walletFile) {
-            return InitError(strprintf(_("Error loading wallet %s. -wallet parameter must only specify a filename (not a path)."), walletFile));
-        }
+    // -salvagewallet can only be used when loading a single wallet
+    bool salvage_wallet = gArgs.GetBoolArg("-salvagewallet", false) && wallet_files.size() <= 1;
```

Comment:
Throw an error if `-salvagewallet` an there are multiple wallets?

Reply:
Sounds like a sensible change in behaviour. I think it's beyond the scope of this PR.

Reply:
Updated comment as suggested by @ryanofsky .

Reply:
Appears this comment update was lost somewhere?